src/httpd_dacp.c

```diff
@@ -410,8 +411,9 @@ dacp_queueitem_add(const char *query, const char *queuefilter, const char *sort,
     }
 
   player_get_status(&status);
-
-  if (mode == 3)
+  count = 0;
+  db_queue_get_count(&count);
+  if (mode == 3 && count > 0)
     ret = db_queue_add_by_queryafteritemid(&qp, status.item_id);
```

Comment:
when queue is empty, `status.item_id ==0` which requires us to`db_queue add_by_query()`